crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -948,9 +990,90 @@ mod tests {
                 log_dir: Default::default(),
             },
             "testDevice",
+            std::path::Path::new(""),
+            &mut std::collections::HashMap::default(),
         )
         .await
         .unwrap();
         assert_eq!(output, vec![]);
     }
+
+    /// Creates `n` child devices named "child-n".
+    /// for each child device a `k` is selected using a random seed so that
+    /// each child devices is assigned a random set of operations.
+    ///
+    /// The resulting dir structure is the following:
+    /// .
+    /// └── operations
+    ///     └── c8y
+    ///         ├── child-0
+    ///         │   └── c8y_LogfileRequest
+    ///         ├── child-1
+    ///         │   ├── c8y_Command
+    ///         │   ├── c8y_DownloadConfigFile
+    ///         │   └── c8y_SoftwareUpdate
+    ///         ├── child-2
+    ///         │   ├── c8y_Command
+    ///         │   ├── c8y_DownloadConfigFile
+    ///         │   └── c8y_LogfileRequest
+    ///         └── child-3
+    ///             └── c8y_LogfileRequest
+    fn make_n_child_devices_with_k_operations(n: u8, ttd: &TempTedgeDir) {
+        let mut rng = rand::rngs::StdRng::seed_from_u64(11);
+        let dist = rand::distributions::Uniform::from(1..OPERATIONS.len());
+
+        for i in 0..n {
+            let dir = ttd.dir(&format!("child-{i}"));
+
+            let k = dist.sample(&mut rng);
+            let operations: Vec<_> = OPERATIONS
+                .choose_multiple(&mut rand::thread_rng(), k)
+                .collect();
+            for op in &operations {
+                dir.file(op);
+            }
+        }
+    }
+
+    /// Tests that the child device cache is updated and that only devices represented locally are
+    /// actually updated to the cloud.
+    ///
+    /// This means that:
+    ///     - Any child device that is not present locally but is seen in the cloud, will not be
+    ///     updated with operations. This child device will not be cached.
+    ///
+    ///     - Any child device that is present locally but not in the cloud will be created and
+    ///     then supported operations will be published to the cloud and the device will be cached.
```

Comment:
> then supported operations will be published to the cloud

This part is not tested. Neither the messages returned by `process_smartrest`.

For that test to be perfect, I would add a check that there is a declaration message for each child-device known locally but not in the cloud.